pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,24 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
```

Comment:
I think you're right. That race already exists though, correct? At least by immediately adding the pod information here, we have a shot at avoiding it. If we do hit the race, worse case is it will detach then re-attach when the pods are processed.